### PR TITLE
added tests and implementation for options object

### PR DESCRIPTION
**Pull request by carycheng**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @carycheng, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: src/main/java/com/box/sdk/MetadataTemplate.java [hunk 1]
```diff
@@ -685,7 +700,14 @@ public void setDescription(String description) {
          * @param options list of possible options for enum type of the field.
          */
         public void setOptions(List<String> options) {
-            this.options = options;
+            JsonObject optionObject = new JsonObject();
```
Comment (mattwiller) [inline src/main/java/com/box/sdk/MetadataTemplate.java]: Is there any issue with re-using the same `JsonObject` for each key here?

File: src/test/java/com/box/sdk/MetadataTemplateTest.java [hunk 2]
```diff
@@ -53,7 +55,7 @@ public void testDeleteMetadataTemplateSucceeds() {
     @Test
     @Category(IntegrationTest.class)
     public void createMetadataTemplateSucceeds() {
-        BoxAPIConnection api = new BoxAPIConnection(TestConfig.getAccessToken());
+        BoxAPIConnection api = new BoxAPIConnection("UXu8HsUlB4K5uDWGWSO075k1qq9hle93");
```
Comment (mattwiller) [inline src/test/java/com/box/sdk/MetadataTemplateTest.java]: Don't commit this!

File: src/test/java/com/box/sdk/MetadataTemplateTest.java [hunk 3]
```diff
@@ -78,6 +80,7 @@ public void createMetadataTemplateSucceeds() {
             MetadataTemplate template = MetadataTemplate.createMetadataTemplate(api, "enterprise",
                     "documentFlow03", "Document Flow 03", false, fields);
         } catch (BoxAPIException apiEx) {
+            System.out.println("Exception: " + apiEx);
```
Comment (mattwiller) [inline src/test/java/com/box/sdk/MetadataTemplateTest.java]: Don't commit this either

**Pull request merged**